### PR TITLE
Fix refactor validation state

### PR DESCRIPTION
**Pull request by Karim94**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/8165/mobile-wrong-validation-state-after-disconnect
#### Testing instructions  
Disconnect device during "sign up user info" step. Any text inputs which have not been validated should not give any errors. On reconnect, the inputs are re-validated, and any invalid inputs would show errors.

Note: During implementation I also tested the case where a text input has the "alwaysDirty" prop, which currently none of our inputs have.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: app/components/shared/styled-text-input.js [hunk 1]
```diff
@@ -36,6 +38,13 @@ export default class StyledTextInput extends SafeComponent {
         this.focusedAnim = new Animated.Value(0);
     }
 
+    componentDidMount() {
+        this.reaction = reaction(() => socket.connected, () => {
```
Comment (seavan) [inline app/components/shared/styled-text-input.js]: don't forget to free reactions after dismount.

Comment (Karim94) [reply app/components/shared/styled-text-input.js]: fixed

File: app/components/shared/styled-text-input.js [hunk 2]
```diff
@@ -36,10 +38,18 @@ export default class StyledTextInput extends SafeComponent {
...
+            // Only run validation on reconnect, not on disconnect
+            if (socket.connected) this.validate();
+        }, true);
+    }
+
     componentWillUnmount() {
         if (uiState.focusedTextBox === this.textInput) {
             uiState.focusedTextBox = null;
         }
+        this.reaction = null;
```
Comment (seavan) [inline app/components/shared/styled-text-input.js]: @Karim94 ????

**Comment by ebilg** [comment]
can be merged
